app/assets/javascripts/shopify_app/top_level_interaction.js

```diff
@@ -0,0 +1,21 @@
+(function() {
+  function setUpTopLevelInteraction() {
+    var TopLevelInteraction = new ITPHelper({
+      content: '#TopLevelInteractionContent',
+      action: '#TopLevelInteractionButton',
+      redirectUrl: window.redirectUrl
+    });
+
+    if (!TopLevelInteraction.itpContent) {
+      return;
+    }
+
+    if (TopLevelInteraction.userAgentIsAffected()) {
+      TopLevelInteraction.setUpContent();
+    } else {
+      TopLevelInteraction.redirect();
+    }
```

Comment:
I think we could move everything from line 9 to here into `ITPHelper` itself, similar to `execute` in `StorageAccessHelper`, and just have DOMContentLoaded call it. I think it would make it easier to follow what's happening.